plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -367,13 +380,13 @@ protected boolean isSupportedJoinCondition(JdbcJoinCondition joinCondition)
         return joinCondition.getOperator() != JoinCondition.Operator.IS_DISTINCT_FROM;
     }
 
-    public static LongWriteFunction oracleDateWriteFunction()
+    public static LongWriteFunction trinoDateToOracleDateWriteFunction()
     {
         return new LongWriteFunction() {
             @Override
             public String getBindExpression()
             {
-                return "TO_DATE(?, 'SYYYY-MM-DD')";
+                return "TO_DATE(?, 'SYYYY-MM-DD HH24:MI:SS')";
```

Comment:
if we simply bind `LocalDate` (with no bind expression), what happens then? 
sounds like the code was introduced for dates before 1582-10-14, but no code comment indicates that

Reply:
as above, use `SYYYY-MM-DD` pattern (sans hour portion), and revert `DATE_FORMATTER`

Reply:
Oracle JDBC driver accepts binding the object, but mapping tests would fail with:
```
java.lang.AssertionError: [Rows for query [SELECT 'found' FROM test_datezgyekqtfkk WHERE col_4 IS NOT DISTINCT FROM TIMESTAMP '1970-01-01 00:00:00']] 
Expecting:
  <>
to contain exactly in any order:
  <[(found)]>
but could not find the following elements:
  <(found)>
```